grpc-protocol/src/main/java/com/linecorp/armeria/internal/common/grpc/protocol/GrpcTrailersUtil.java

```diff
@@ -43,12 +44,17 @@ public final class GrpcTrailersUtil {
      * as {@code true}.
      */
     public static void addStatusMessageToTrailers(
-            HttpHeadersBuilder trailersBuilder, int code, @Nullable String message) {
+            HttpHeadersBuilder trailersBuilder, int code, @Nullable String message,
+            @Nullable byte[] details) {
         trailersBuilder.endOfStream(true);
         trailersBuilder.add(GrpcHeaderNames.GRPC_STATUS, StringUtil.toString(code));
         if (message != null) {
             trailersBuilder.add(GrpcHeaderNames.GRPC_MESSAGE, StatusMessageEscaper.escape(message));
         }
+        if (details != null && details.length > 0) {
+            final String encodedDetails = Base64.getEncoder().encodeToString(details);
```

Comment:
Is this a spec Base64 encoding for the details?

Reply:
Only a header that ends with `-bin` is base64-encoded.
https://github.com/grpc/grpc-java/blob/50cdfa9f05131ef0066cca99140a5f99339a49e3/api/src/main/java/io/grpc/Metadata.java#L64-L66

Reply:
I see, I wasn't aware of this.
Then should we also append the header name with `-bin`? I don't think we can guarantee that the input will be ASCII.


![Screen Shot 2022-07-21 at 11 35 17 PM](https://user-images.githubusercontent.com/8510579/180240784-3aa94cb0-d807-46ee-849f-a1edb52ad20a.png)
https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md

Reply:
I think so. I proposed a new header name on https://github.com/line/armeria/pull/4341#discussion_r926588353